src/services/formatting/formattingScanner.ts

```diff
@@ -126,7 +126,7 @@ namespace ts.formatting {
         }
 
         function shouldRescanJsxText(node: Node): boolean {
-            return isJsxText(node);
+            return isJsxText(node) || isJsxElement(node) && lastTokenInfo?.token.kind === SyntaxKind.JsxText;

```

Comment:
Do you recall what the scanner was doing wrong without this? This stuff is so hard to reason about; I want to have at least some understanding of the problem before I approve 😅

Reply:
While debugging this problem I noticed that

```ts
<div>' <span></span></div>
```

`formattingScanner` tries to scan the string literal after `'`

```ts
' <span></span></div>
```

However, it is part of the `JsxText`, so when the `scanner` encounters the next child element in the `JsxText` context, it should rescan the `JsxText`. Maybe we can use `advance()` with other conditions to skip scanning the string literals.

Reply:
Yeah, so I’m curious if other interruptions to JsxText after a quote would present a problem too, like

```
<div>' {foo}</div>
```